src/App/ExternalServices/HDXInterface.php

```diff
@@ -0,0 +1,159 @@
+            $config = [
+                'base_uri' => $this->ckanURL,
+                'headers' => ['Authorization' => $this->userAPIKey],
+            ];
+            //if we passed in a mock handler
+            if (isset($this->handler)) {
+                $config['handler'] = $this->handler;
+            }
+            //create a Guzzle client, used by CKAN api
+            $client = new Client($config);
+            $this->ckanClient = new CkanApiClient($client);
+        }
+        return $this->ckanClient;
+    }
+
+    // returns ID or null
+    public function getDatasetIDByTitle(string $titleText)
+    {
+        /// setup a search query by title
+        $data = [
+           'q' => 'title:'.$titleText
+        ];
+        $datasetId = null;
+        try {
+            $allMatchingDatasets = $this->getApiClient()->dataset()->all($data);
+        } catch (Exception $e) {
+            Log::error('Unable to find HDX datasets by title '.print_r($e, true));
+        }
+        if ($allMatchingDatasets && array_key_exists('result', $allMatchingDatasets)) {
+            if ($allMatchingDatasets['result']['count'] > 1) {
+                Log::debug('Multiple datasets found: '.print_r($allMatchingDatasets['result']['count'], true));
+            }
+            foreach ($allMatchingDatasets['result']['results'] as $eachDataset) {
+                $datasetId = $eachDataset['id'];
+            }
+        }
+        return $datasetId;
+    }
+
+    /** Note: if error condition is the result, then we ignore it gracefully,
+    * but the full error response array will be returned instead of a confirmation array
+    */
+    public function updateHDXDatasetRecord(array $metadata)
+    {
+        $apiClient = $this->getApiClient();
+        $updateResult = [];
+        try {
+            $updateResult = $apiClient->dataset()->update($metadata);
+        } catch (Exception $e) {
+            // @TODO: be graceful here
+            $updateResult = ['error' => 'Unable to update dataset on HDX server.'];
+            Log::error('Unable to update dataset on HDX server: '.print_r($e, true));
+        }
+        return $updateResult;
+    }
+
+    /** Note: if error condition is the result, then we ignore it gracefully,
+    * but the full error response array will be returned instead of a confirmation array
+    */
+    public function createHDXDatasetRecord(array $metadata)
+    {
+        $apiClient = $this->getApiClient();
+        $createResult = [];
+        try {
+            $createResult = $apiClient->dataset()->create($metadata);
+        } catch (Exception $e) {
+            // @TODO: be graceful here
+            $createResult = ['error' => 'Unable to create dataset on HDX server.'];
+            Log::error('Unable to create dataset on HDX server: '.print_r($e, true));
+        }
+        return $createResult;
+    }
+
+    public function getAllOrganizationsForUser()
+    {
+        $apiClient = $this->getApiClient();
```

Comment:
FIXING this now to make the get_organizations_for_user request which is the one hdx has available to get a specific list of organisations for this user api key

Reply:
@rowasc - oops, sorry if that wasn't the right API call. Let me know if you want me to grab this and/or update the tests.